src/main.c

```diff
@@ -1132,28 +1136,38 @@ void handleMod3Key(KBDLLHOOKSTRUCT keyInfo, bool isKeyUp) {
 	if (isKeyUp) {
 		if (keyInfo.scanCode == scanCodeMod3R) {
 			level3modRightPressed = false;
+			double level3modRightPressedDuration = (double) (clock() - level3modRightPressedInstant) / CLOCKS_PER_SEC;
 			modState.mod3 = level3modLeftPressed | level3modRightPressed;
-			if (mod3RAsReturn && level3modRightAndNoOtherKeyPressed) {
+			if (mod3RAsReturn && level3modRightAndNoOtherKeyPressed && level3modRightPressedDuration < 0.5) {
 				sendUp(keyInfo.vkCode, keyInfo.scanCode, false); // release Mod3_R
 				sendDownUp(VK_RETURN, 28, true); // send Return
 				level3modRightAndNoOtherKeyPressed = false;
 			}
 		} else { // scanCodeMod3L (CapsLock)
 			level3modLeftPressed = false;
+			double level3modLeftPressedDuration = (double) (clock() - level3modLeftPressedInstant) / CLOCKS_PER_SEC;
 			modState.mod3 = level3modLeftPressed | level3modRightPressed;
-			if (capsLockAsEscape && level3modLeftAndNoOtherKeyPressed) {
+			if (capsLockAsEscape && level3modLeftAndNoOtherKeyPressed && level3modLeftPressedDuration < 0.5) {
 				sendUp(VK_CAPITAL, 58, false); // release Mod3_R
 				sendDownUp(VK_ESCAPE, 1, true); // send Escape
 				level3modLeftAndNoOtherKeyPressed = false;
 			}
 		}
 	} else { // keyDown
 		if (keyInfo.scanCode == scanCodeMod3R) {
-			level3modRightPressed = true;
+			if (!level3modRightPressed)
+			{
```

Comment:
Could you move the curly brace into the line above for being consistent with the rest of the code?